src/components/reusable/daterangepicker/daterangepicker.ts

```diff
@@ -0,0 +1,288 @@
+  regexDateFormat = /^\d{4}-\d{2}-\d{2}$/;
+
+  /**
+   * Attached internals for form association.
+   * @ignore
+   */
+  @state()
+  internals = this.attachInternals();
+
+  /** Optional text beneath the input. */
+  @property({ type: String })
+  caption = '';
+
+  /** Datepicker size. "sm", "md", or "lg". */
+  @property({ type: String })
+  size = 'md';
+
+  /** Datepicker Start date in YYYY-MM-DD format. */
+  @property({ type: String })
+  startDate = '';
+
+  /** Datepicker End date in YYYY-MM-DD format. */
+  @property({ type: String })
+  endDate = '';
+
+  /** Datepicker name. */
+  @property({ type: String })
+  name = '';
+
+  /** Makes the date required. */
+  @property({ type: Boolean })
+  required = false;
+
+  /** Date disabled state. */
+  @property({ type: Boolean })
+  disabled = false;
+
+  /** Date invalid text. */
+  @property({ type: String })
+  invalidText = '';
+
+  /** Date warning text */
+  @property({ type: String })
+  warnText = '';
+
+  /** Maximum date in YYYY-MM-DD format.
+   * If the value isn't a possible date string in the format, then the element has no maximum date value
+   */
+  @property({ type: String })
+  maxDate = '';
+
+  /** Minimum date in YYYY-MM-DD format,
+   * If the value isn't a possible date string in the format, then the element has no minimum date value.
+   */
+  @property({ type: String })
+  minDate = '';
+
+  /** Specifies the granularity that the value must adhere to, or the special value any,
+   * For date inputs, the value of step is given in days; and is treated as a number of milliseconds equal to 86,400,000 times the step value.
+   * The default value of step is 1, indicating 1 day.*/
+  @property({ type: String })
+  step = '';
+
+  override render() {
+    return html`
+      <label
+        class="datepicker-label-text"
+        for=${this.name}
+        ?disabled=${this.disabled}
+      >
+        ${this.required ? html`<span class="required">*</span>` : null}
+        <slot></slot>
+      </label>
+
+      <div class="input-wrapper">
+        <input
+          class="${classMap({
+            'size--sm': this.size === 'sm',
+            'size--lg': this.size === 'lg',
+          })}"
+          type="date"
+          id=${this.name ? `${this.name}-1` : 'date-range-1'}
+          name=${this.name}
+          value=${this.startDate}
+          ?required=${this.required}
+          ?disabled=${this.disabled}
+          ?invalid=${this.invalidText !== ''}
+          min=${ifDefined(this.minDate)}
+          max=${ifDefined(this.maxDate)}
+          step=${ifDefined(this.step)}
+          @input=${(e: any) => this.handleStartDate(e)}
+        />
+      </div>
+
+      <span class="range-span">—</span>
+      <div class="input-wrapper">
+        <input
+          class="${classMap({
+            'size--sm': this.size === 'sm',
+            'size--lg': this.size === 'lg',
+          })}"
+          type="date"
+          id=${this.name ? `${this.name}-2` : 'date-range-2'}
```

Comment:
Same comment, `name` and `id` = `this.name + '-end`.

Reply:
done